### PR TITLE
Transactionize the analysis and job objects

### PR DESCRIPTION
**Pull request by josenavas**
The analysis and job object now use transactions.

I've been removing some SQL queries to use `convert_to_id` in a couple of places.

### CODE REVIEW

**Comment by josenavas** [comment]
This have the same failure as https://github.com/biocore/qiita/pull/1326
Once it gets merged I will pull from upstream/transaction and the failure will go away, so this is safe to review

**Comment by antgonza** [comment]
This looks fine. :+1:

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -100,11 +100,14 @@ def get_by_status(cls, status):
         set of int
             All analyses in the database with the given status
         """
-        conn_handler = SQLConnectionHandler()
-        sql = ("SELECT analysis_id FROM qiita.{0} a JOIN qiita.{0}_status ans "
-               "ON a.analysis_status_id = ans.analysis_status_id WHERE "
-               "ans.status = %s".format(cls._table))
-        return {x[0] for x in conn_handler.execute_fetchall(sql, (status,))}
+        with TRN:
+            sql = """SELECT analysis_id
```
Comment (squirrelo) [inline qiita_db/analysis.py]: This indentation is odd, and this SQL should also be changed to use `USING`. Standard SQL indentation is:
```SQL
SELECT analysis_id  FROM qiita.{0}
JOIN qiita.{0}_status USING (analysis_status_id)
WHERE ans.status = %s
```

File: qiita_db/analysis.py [hunk 2]
```diff
@@ -363,13 +379,17 @@ def data_types(self):
...
         """
-        sql = ("SELECT DISTINCT data_type from qiita.data_type d JOIN "
-               "qiita.processed_data p ON p.data_type_id = d.data_type_id "
-               "JOIN qiita.analysis_sample a ON p.processed_data_id = "
-               "a.processed_data_id WHERE a.analysis_id = %s ORDER BY "
-               "data_type")
-        conn_handler = SQLConnectionHandler()
-        return [x[0] for x in conn_handler.execute_fetchall(sql, (self._id, ))]
+        with TRN:
+            sql = """SELECT DISTINCT data_type
```
Comment (squirrelo) [inline qiita_db/analysis.py]: Same as above for indentation and use of `USING`

File: qiita_db/analysis.py [hunk 3]
```diff
@@ -393,28 +414,25 @@ def all_associated_filepath_ids(self):
...
-
-        job_filepaths = {row[0]
-                         for row in conn_handler.execute_fetchall(sql,
-                                                                  [self._id])}
-
-        filepaths = filepaths.union(job_filepaths)
-
-        return filepaths
+        with TRN:
+            sql = """SELECT f.filepath_id
```
Comment (squirrelo) [inline qiita_db/analysis.py]: Same here and the SQL on line 426

File: qiita_db/analysis.py [hunk 4]
```diff
@@ -425,20 +443,24 @@ def biom_tables(self):
...
-        if not tables:
-            return {}
-        ret_tables = {}
-        _, base_fp = get_mountpoint(self._table)[0]
-        for fp in tables:
-            ret_tables[fp[0]] = join(base_fp, fp[1])
-        return ret_tables
+        with TRN:
+            fptypeid = convert_to_id("biom", "filepath_type")
+            sql = """SELECT dt.data_type, f.filepath
```
Comment (squirrelo) [inline qiita_db/analysis.py]: Same here

File: qiita_db/analysis.py [hunk 5]
```diff
@@ -449,43 +471,65 @@ def mapping_file(self):
...
-               "WHERE af.analysis_id = %s AND f.filepath_type_id = %s")
-        mapping_fp = conn_handler.execute_fetchone(sql, (self._id, fptypeid))
-        if not mapping_fp:
-            return None
-
-        _, base_fp = get_mountpoint(self._table)[0]
-        return join(base_fp, mapping_fp[0])
+        with TRN:
+            fptypeid = convert_to_id("plain_text", "filepath_type")
+            sql = """SELECT f.filepath
```
Comment (squirrelo) [inline qiita_db/analysis.py]: Ok there's a lot of `USING` that needs to be done. Last comment on it.

**Comment by squirrelo** [comment]
Some SQL cleanup that needs to be done (not your fault but good to do since this pull request is going in to clean things up anyway). Other than that, looks good.

File: qiita_db/analysis.py (hunk 1 above)
Comment (josenavas) [inline qiita_db/analysis.py]: @squirrelo can you point to the "standard" that you're talking about? There is no standard, only coding guidelines (that we can user or not) but None of the ones that I've seen use your indentation. Usually is common practice to put the FROM in the next line under the SELECT, not following it (quick googling will show this to you...)

I can agree on adding the JOIN in a single line but it didn't fit. Will change for USING.

File: qiita_db/analysis.py (hunk 2 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/analysis.py (hunk 3 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/analysis.py (hunk 4 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

File: qiita_db/analysis.py (hunk 5 above)
Comment (josenavas) [inline qiita_db/analysis.py]: Done

**Comment by josenavas** [comment]
@squirrelo  comments addressed

**Comment by squirrelo** [comment]
:+1:

**Pull request merged**